### PR TITLE
Impower (Verified Publisher) - Update file download definition so operation works within flows

### PR DESCRIPTION
**Pull request by albertmatyi**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- ~[x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)~
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
  
image

- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
  
image

~- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.~

### CODE REVIEW

**Comment by vmanoharas** [comment]
Hello @albertmatyi, 

I noticed that your connector currently in partner verification stage with new version in ISV studio.  Would you like merge this change with existing version or upload into ISV studio as new version once I approved it. Please advise.

**Comment by albertmatyi** [comment]
Hey @vmanoharas ,

I pushed these changes because we have only noticed the filedownload issues in the ISV studio as we were starting building flows.

We tested now more thoroughly (via custom connector based flows) and that's why we are pushing this update.

Once through, we wish to build a flow in the sandbox environment in ISV.

In other words: This MR is required to proceed with the approval.

**Comment by vmanoharas** [comment]
Hello @albertmatyi, 

Thank you for the clarification. 

However, there are some swagger validation error. Could you please review below errors and confirm that the existing flow will not break. 

##[error]A produces option has been removed from the operation. ValuePath: operation/downloadDocumentsByFilterUsingGET/produces Source: NewSwagger
##[error]The type has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/downloadDocumentsByFilterUsingGET/responses/200/schema/type Source: NewSwagger

**Comment by albertmatyi** [comment]
Hello @vmanoharas ,

The response mime-types (produces option) has been updated to "*/*" on file download endpoints since those endpoints can produce any mimetype.
Thus: I confirm that that change is correct.

I've also added a webhook trigger to the MR 

Here is a successful flow built with the webhook and successful download commands (tested both of them)

image

**Comment by vmanoharas** [comment]
Thank you very much @albertmatyi for confirming.

File: certified-connectors/Impower/apiDefinition.swagger.json
Comment (vmanoharas) [inline certified-connectors/Impower/apiDefinition.swagger.json:247]: Can you please add description property? 

Please refer the schema and template for the help.

Error - ##[error]The 'description' property is required. ValuePath: /paths/~1v2~1connections~1{connectionIdX}/put

Schema - https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/apiDefinition.swagger.schema.json

Template - https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

File: certified-connectors/Impower/apiDefinition.swagger.json
Comment (vmanoharas) [inline certified-connectors/Impower/apiDefinition.swagger.json]: Can you please reduce the characters? the swagger validation triggered below error. 

##[error]The 'x-ms-summary' value goes over the character limit '80' ValuePath: /paths/~1v2~1connections/post/parameters/0/x-ms-summary

**Review by vmanoharas — COMMENTED** [review]
Hi @albertmatyi, Can you please resolve review comments?

File: certified-connectors/Impower/apiDefinition.swagger.json
Comment (albertmatyi) [reply certified-connectors/Impower/apiDefinition.swagger.json:247]: added description property

File: certified-connectors/Impower/apiDefinition.swagger.json
Comment (albertmatyi) [reply certified-connectors/Impower/apiDefinition.swagger.json]: reduced description length.

**Comment by albertmatyi** [comment]
Hey @vmanoharas : 

1. reran paconn validation - passed
2. tested connector in flow - passed

**Review by vmanoharas — APPROVED** [review]
Hello @albertmatyi 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**
